src/utils/gauges.ts

```diff
@@ -82,62 +98,88 @@ export const initialGaugesState: Gauges = {
-      createMultiCallContract<GaugeHelperContract>(
-        gaugeHelperContractAddress,
-        GAUGE_HELPER_CONTRACT_ABI,
-      )
-
     const gaugeControllerMultiCall = createMultiCallContract<GaugeController>(
       gaugeControllerContractAddress,
       GAUGE_CONTROLLER_ABI,
     )
 
+    // sidechain contracts
+
+    const childGaugeFactoryMultiCall =
+      createMultiCallContract<ChildGaugeFactory>(
+        childGaugeFactoryAddress,
+        CHILD_GAUGE_FACTORY_ABI,
+      )
+
+    /* ------- end of base contracts instantiation -------  */
+
     const gaugeAddresses: string[] = (
       await ethCallProvider.all(
-        enumerate(gaugeCount, 0).map((value) =>
-          gaugeControllerMultiCall.gauges(value),
+        enumerate(gaugeCount, 0).map(
+          retrieveGaugeAddresses(
+            gaugeControllerMultiCall,
+            childGaugeFactoryMultiCall,
+          ),
         ),
       )
     ).map((address) => address.toLowerCase())
 
-    const gaugeRewardsPromise = ethCallProvider.all(
-      gaugeAddresses.map((address) =>
-        gaugeHelperContractMultiCall.getGaugeRewards(address),
-      ),
-    )
-    const gaugeWeightsPromise: Promise<BigNumber[]> = ethCallProvider.all(
-      gaugeAddresses.map((gaugeAddress) =>
-        gaugeControllerMultiCall.get_gauge_weight(gaugeAddress),
-      ),
+    const gaugeMulticallContracts = retrieveGaugeContracts(
+      gaugeAddresses,
+      chainId,
     )
-    const gaugeRelativeWeightsPromise: Promise<BigNumber[]> =
-      ethCallProvider.all(
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-        gaugeAddresses.map((gaugeAddress) =>
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call
-          gaugeControllerMultiCall.gauge_relative_weight(gaugeAddress),
-        ),
-      )
 
-    const gaugeMulticallContracts = gaugeAddresses.map((gaugeAddress) =>
-      createMultiCallContract<LiquidityGaugeV5>(
-        gaugeAddress,
-        LIQUIDITY_GAUGE_V5_ABI,
-      ),
-    )
+    const gaugeWeightsPromise: Promise<BigNumber[]> =
+      chainId === ChainId.MAINNET
+        ? ethCallProvider.all(
+            gaugeAddresses.map((gaugeAddress) =>
+              gaugeControllerMultiCall.get_gauge_weight(gaugeAddress),
+            ),
+          )
+        : Promise.resolve(Array<BigNumber>(gaugeAddresses.length).fill(Zero))
+
+    const gaugeRelativeWeightsPromise: Promise<BigNumber[]> =
+      chainId === ChainId.MAINNET
+        ? ethCallProvider.all(
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+            gaugeAddresses.map((gaugeAddress) =>
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call
+              gaugeControllerMultiCall.gauge_relative_weight(gaugeAddress),
+            ),
+          )
+        : Promise.resolve(Array<BigNumber>(gaugeAddresses.length).fill(Zero))
```

Comment:
Looks good for mainnet calls.
Just wondering whats the implication of being on sidechain and this promise returning an array with 0s?

Reply:
@penandlim  Since we're not allowing user to view the veSDL page while on side chains, we don't need to populate the weights. Will go through it again to see if we need the weights and relative weights for anything else.